[AGM-2388] Updated the article in KB about the urltransform modifier example (#675)
AdguardTeam/KnowledgeBase: docs/general/ad-filtering/create-own-filters.md

`$urltransform` rules can also be disabled by `$document` and `$urlblock` exception rules. But basic exception rules without modifiers do not do that. For example, `@@||example.com^` will not disable `$urltransform=/X/Y/` for requests to **example.com**, but `@@||example.com^$urlblock` will.

> **The rule example on how to clean affiliate links**

Reviewer: ```suggestion
**The rule example for cleaning affiliate links**
```

> Many websites use tracking URLs to monitor clicks before redirecting to the actual destination. These URLs include marketing parameters and analytics tokens, which we can remove to improve privacy.

Reviewer: ```suggestion
Many websites use tracking URLs to monitor clicks before redirecting to the actual destination. These URLs contain marketing parameters and analytics tokens that can be removed to improve privacy.
```

> Below is an example of how to “clean” the original link to bypass tracking websites and go directly to the destination.

Author: мне кажется, что со словом clean понятнее, что мы делаем

> Below is an example of how to “clean” the original link to bypass tracking websites and go directly to the destination.

Author: можно просто кавычки поменять

> Below is an example of how to “clean” the original link to bypass tracking websites and go directly to the destination.

Reviewer 2: Обновил предложенный вариант (`how to obtain the clean destination link`). Лучше?)

> Below you'll find an example demonstrating how to "clean" the original link to bypass tracking websites and proceed directly to the target destination.

Reviewer: ```suggestion
Below is an example of how to “clean” the original link to bypass tracking websites and go directly to the destination.
```

In our example:

>  1. Original url (messy tracking link): `https://www.aff.example.com/visit?url=https%3A%2F%2Fwww.somestore.com%2F%26referrer%3Dhttps%3A%2F%2Fwww.aff.example.com%2F%26ref%3Dref-123`

Reviewer: ```suggestion
 1. Original URL (dirty tracking link): `https://www.aff.example.com/visit?url=https%3A%2F%2Fwww.somestore.com%2F%26referrer%3Dhttps%3A%2F%2Fwww.aff.example.com%2F%26ref%3Dref-123`
```

>  1. Tracking websites' “clean” url: `https://www.aff.example.com/visit?url=https://www.somestore.com/`

Reviewer: ```suggestion
 1. Clean URL of a tracking website: `https://www.aff.example.com/visit?url=https://www.somestore.com/`
```

>  1. The website you want to visit: `https://www.somestore.com`

Reviewer: да и так понятно вроде

>  1. The destination website you intend to visit: `https://www.somestore.com`

Reviewer: ```suggestion
 1. The website you want to visit: `https://www.somestore.com`
```

> To achieve the “clean” url result we need to decode url-encoded characters (like `%3A` → `:`, `%2F` → `/`, etc), and extract the real url from the tracking parameters. We will use the `$urltransform` modifier to do it. The following 4 rules replace url-encoded symbols with their actual characters:

Reviewer 2: ```suggestion
To clean the URL, we first need to decode special characters (like `%3A` → `:`, `%2F` → `/`, etc), and extract the real url from the tracking parameters. We will use the `$urltransform` modifier to do it. The following 4 rules replace URL-encoded symbols with their actual characters:
```

> To clean the URL, we first need to decode special characters (like `%3A` → `:`, `%2F` → `/`, etc), and extract the real url from the tracking parameters. We will use the `$urltransform` modifier to do it. The following 4 rules replace URL-encoded symbols with their actual characters:

Reviewer: ```suggestion
To clean the URL, we first need to decode special characters (like `%3A` → `:`, `%2F` → `/`, etc.) and extract the real URL from the tracking parameters. We will use the `$urltransform` modifier to do i.his The following 4 rules replace URL-encoded symbols with their real characters:
```

`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/%3A/:/`
`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/%2F/\//`
`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/%3F/?/`
`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/%3D/=/`
`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/%26/&/`

> After that, we need to write the rule that will block the tracking website and transfer you directly to the targeted website (somestore.com):

Reviewer: ```suggestion
After that, we need to write the rule that will block the tracking website and redirect you directly to the target website (somestore.com):
```

`/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com\/visit\?url=/$urltransform=/^https?:\/\/(?:[a-z0-9-]+\.)*?aff\.example\.com.*url=([^&]*).*/\$1/`

> Done! Now you can bypass tracking mechanisms and maintain full privacy protection.

Reviewer 2: ```suggestion
Tracking links will now be automatically cleaned, enabling direct navigation to the destination site without tracking.
```

> Tracking links will now be automatically cleaned, enabling direct navigation to the destination site without tracking.

Reviewer: ```suggestion
Tracking links will now be automatically cleaned up, allowing direct navigation to the destination website without tracking.
```
